docs/docker_setup_guide.md

```diff
@@ -0,0 +1,204 @@
+# Medispeak Developer Setup Guide with Docker Compose
+
+This guide will walk you through the steps to set up the Medispeak backend using Docker Compose. The configuration includes two main services:
+
+1. **Rails Backend** (`medispeak_backend`): The main application container running the Medispeak Rails backend.
+2. **PostgreSQL Database** (`medispeak_db`): A database container that handles data persistence for the Medispeak application.
+
+---
+
+## Prerequisites
+
+Before getting started, make sure you have the following installed on your local machine:
+
+1. **Docker**: For containerizing and running the services.
+2. **Docker Compose**: To manage the multi-container setup.
+
+Ensure Docker and Docker Compose are properly installed and running. You can verify the installation by running:
+
+```bash
+docker --version
+docker-compose --version
+```
+
+## 1. Docker Compose Configuration Overview
+
+The `docker-compose.yml` file defines two main services: the PostgreSQL database (`medispeak_db`) and the Rails backend (`medispeak_backend`).
+
+### 1.1 PostgreSQL Database Configuration
+
+The `medispeak_db` service uses the official PostgreSQL image and specifies several key configurations:
+
+- **Environment Variables**:
+  - `POSTGRES_DB`: Defines the database name.
+  - `POSTGRES_USER`: Sets the username for the database.
+  - `POSTGRES_PASSWORD`: Sets the password for the PostgreSQL user.
+- **Ports**: The container exposes the PostgreSQL port defined by `DB_PORT`.
+- **Volumes**: Data is persisted across container restarts using a Docker volume (`medispeak_data`).
+
+### 1.2 Medispeak Backend Configuration
+
+The `medispeak_backend` service is the core Rails application. It depends on the `medispeak_db` service being healthy before starting. Key aspects:
+
+- **Build Arguments**: The build process takes several arguments, such as `RAILS_ENV`, `BUNDLE_DEPLOYMENT`, and others for bundling gems and preparing the Rails environment.
+- **Environment Variables**:
+  - OpenAI credentials (`OPENAI_ACCESS_TOKEN`, `OPENAI_ORGANIZATION_ID`).
+  - AWS credentials for file storage (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, etc.).
+  - PostgreSQL database connection details (`DB_HOST`, `DB_NAME`, `DB_USERNAME`, `DB_PASSWORD`).
+  - Plugin-related configuration (`PLUGIN_BASE_URL`).
+- **Ports**: The Rails backend exposes the port defined by `BACKEND_PORT`.
+- **Volumes**: The application code is mounted inside the container using a bind mount to keep it in sync with your local development files.
+
+---
+
+## 2. Setting Up Environment Variables
+
+Before running the services, you’ll need to configure environment variables. These variables control the behavior of both the Rails backend and PostgreSQL database.
+
+1. **Create a `.env` file** in the root directory (where the `docker-compose.yml` file is located) with the following content:
+
+```bash
```

Comment:
This is not in sync with `.example.env` example: `POSTGRES_IMAGE_TAG` is missing. Should we keep it in sync or just keep it in `.example.env`

Reply:
@bodhish thanks for the heads up. I have addressed the issue; could you please validate it now?